src/App.tsx

```diff
@@ -1,41 +1,91 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { getUsers } from './api/users';
+import { getPosts, getUserPosts } from './api/posts';
+
 import './App.scss';
 import './styles/general.scss';
+
+import { Loader } from './components/Loader';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsLoading, setPostsLoading] = useState(true);
+
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const fetchUsers = useCallback(async () => {
+    setPostsLoading(true);
+    setUsers(await getUsers());
```

Comment:
better not to do like this, create a separate variable (everywhere)
```suggestion
    const usersFromServer = await getUsers();

    setUsers(usersFromServer);
```